src/pages/home/report/ReportActionsList.tsx

```diff
@@ -267,7 +267,9 @@ function ReportActionsList({
             const isFromNotification = route?.params?.referrer === CONST.REFERRER.NOTIFICATION;
             if ((Visibility.isVisible() || isFromNotification) && scrollingVerticalOffset.current < MSG_VISIBLE_THRESHOLD) {
                 Report.readNewestAction(report.reportID);
-                Navigation.setParams({referrer: undefined});
+                if (isFromNotification) {
+                    Navigation.setParams({referrer: undefined});
+                }
```

Comment:
This also fixes the test of UnreadIndicatorsTest. When the test runner opens the report screen, if it's unread (which is true), it will call both functions above to mark the report as read and remove the `referrer` param. 

When the referrer param is updated, the report screen is re-mounted (on test only), but the readNewestAction optimisticData isn't completed yet and triggers the above logic again, repeatedly, infinitely.

This doesn't happen when we still use withOnyx, specifically the `reportMetadata`. Using `withOnyx`, when the report screen is re-mounted, it's not immediately rendered because `reportMetadata` has a pending merge, giving `optimisticData` of `readNewestAction` time to complete.

I think the best solution is to clear the `referrer` param after the `readNewestAction` `optimisticData` is applied, but I'm not sure how we can do that without using a timeout.

Given the re-mount behavior when updating param is only occurs on test, I updated the code to only clears it when `referrer` exists.